beacon-chain/state/v2/proofs.go

```diff
@@ -0,0 +1,39 @@
+package v2
+
+import (
+	"encoding/binary"
+
+	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+)
+
+const (
+	finalizedRootIndex = 105
+)
+
+// FinalizedRootGeneralizedIndex for the Altair beacon state.
+func FinalizedRootGeneralizedIndex() int {
+	return finalizedRootIndex
+}
+
+// ProveFinalizedRoot crafts a Merkle proof for the finalized root
+// contained within the finalized checkpoint of a beacon state.
+func (b *BeaconState) ProveFinalizedRoot() ([][]byte, error) {
+	cpt := b.state.FinalizedCheckpoint
+	epochBuf := make([]byte, 8)
+	binary.LittleEndian.PutUint64(epochBuf, uint64(cpt.Epoch))
+	epochRoot := bytesutil.ToBytes32(epochBuf)
```

Comment:
The topmost root is derived from epoch number? Interesting

Reply:
no, but the epoch is the neighbor of the finalized root field, so the proof consists of walking up the following branch

```
    state_root
                \
             0x213912
                    \
               htr(finalized_checkpoint)
                        /              \
                  epoch           root
```